app/javascript/controllers/map_controller.js

```diff
@@ -113,22 +140,24 @@ export default class extends Controller {
     const { width: tokenWidth, height: tokenHeight } = tokenImage.getBoundingClientRect()
 
     target.dataset.beingDragged = true
-    target.dataset.offsetX = (tokenWidth / 2) - (clientX - window.scrollX - left)
-    target.dataset.offsetY = (tokenHeight / 2) - (clientY - window.scrollY - top)
+    target.dataset.offsetX = (tokenWidth / 2) - (clientX - left)
+    target.dataset.offsetY = (tokenHeight / 2) - (clientY - top)
 
     event.dataTransfer.setData("text/plain", target.dataset.tokenId)
   }
 
   moveToken(event) {
     event.stopPropagation()
 
-    const { clientX, clientY, target } = event
+    const { target } = event
     const { x: currentX, y: currentY, offsetX, offsetY, tokenId } = target.dataset
-    const { viewportX, viewportY, x: mapX, y: mapY, zoomAmount } = this.imageTarget.dataset
-    const { left: imageLeft, top: imageTop } = this.imageTarget.getBoundingClientRect()
+    const { zoomAmount } = this.imageTarget.dataset
+
+
+    const { x, y }  = this.mapPositionOf(event)
 
-    const newX = Math.round((clientX + parseFloat(offsetX) - (imageLeft + window.scrollX)) / parseFloat(zoomAmount) - (((parseFloat(viewportX) / 2) - parseFloat(mapX)) / parseFloat(zoomAmount)))
-    const newY = Math.round((clientY + parseFloat(offsetY) - (imageTop + window.scrollY)) / parseFloat(zoomAmount) - (((parseFloat(viewportY) / 2) - parseFloat(mapY)) / parseFloat(zoomAmount)))
+    const newX = x + (parseFloat(offsetX) / parseFloat(zoomAmount))
```

Comment:
Just to confirm, why can this, and `moveMap` be changed in this way now?

Reply:
I'm not changing anything in `moveMap` with this PR, but I've extracted the calculations which turn a position in the browser into a position on the map. These can be used for both the token dragging and the pointers.